src/scripts/createLocationAutocompleteData/index.ts

```diff
@@ -0,0 +1,345 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * The data that is being downloaded in this script is provided by
+ * GeoNames (https://www.geonames.org/). We are using the data to compile a set
+ * of US cities and states that match the needs of this project. Their work is
+ * licensed under a Creative Commons Attribution 4.0 License:
+ * https://creativecommons.org/licenses/by/4.0/.
+ *
+ * All database dumps and table definitions can be found here:
+ * https://download.geonames.org/export/dump/.
+ */
+
+import https from "https";
+import {
+  createWriteStream,
+  existsSync,
+  mkdirSync,
+  readFileSync,
+  rmSync,
+  writeFileSync,
+} from "fs";
+import { dirname, resolve } from "path";
+import { fileURLToPath } from "url";
+import AdmZip from "adm-zip";
+
+import {
+  TAlternateNameData,
+  TLocationData,
+  IRelevantLocation,
+  IRelevantLocationAlternate,
+  IDataFileUrls,
+} from "./types.d";
+
+const REMOTE_DATA_URL = "https://download.geonames.org/export/dump";
+const DATA_COUNTRY_CODE = "US";
+const LOCATIONS_DATA_FILE = "locationAutocompleteData.json";
+const FETCH_REMOTE_DATASETS = true;
+const CLEANUP_TMP_DATA_AFTER_FINISHED = true;
+
+// Only include populated place a city, town, village, or other
+// agglomeration of buildings where people live and work.
+// Feature classes and codes: http://www.geonames.org/export/codes.html
+const allowedFeatureClass = "P";
+const allowedFeatureCodes = [
+  "PPL",
+  "PPLA",
+  "PPLA2",
+  "PPLA3",
+  "PPLA4",
+  "PPLC",
+  "PPLL",
+];
+
+function logProgress(currentCount: number, totalCount: number) {
+  const progress = Math.round(((currentCount + 1) / totalCount) * 100);
+  process.stdout.write(
+    `-> ${totalCount}/${currentCount + 1} (${progress}%) \r`
+  );
+}
+
+function writeFromRemoteFile({
+  url,
+  writeStream,
+}: {
+  url: string;
+  writeStream: any;
+}) {
+  return new Promise((resolve, reject) => {
+    https.get(url, (res) => {
+      res.on("end", () => {
+        resolve(res);
+      });
+      res.on("error", (error) => {
+        reject(error);
+      });
+      res.pipe(writeStream);
+    });
+  });
+}
+
+async function fetchRemoteArchive({
+  remoteArchiveUrl,
+  localDownloadPath,
+  localExtractionPath,
+}: IDataFileUrls) {
+  console.info(
+    `Downloading remote file: ${remoteArchiveUrl} -> ${localDownloadPath}`
+  );
+
+  await writeFromRemoteFile({
+    url: remoteArchiveUrl,
+    writeStream: createWriteStream(localDownloadPath),
+  });
+
+  console.info(`Extracting: ${localDownloadPath} -> ${localExtractionPath}`);
+  const zip = new AdmZip(localDownloadPath);
+  zip.extractAllTo(localExtractionPath, true);
+}
+
+try {
+  const startTime = Date.now();
+  console.info("Create autocomplete location data");
+
+  const __dirname = dirname(fileURLToPath(import.meta.url));
+  const tmpDirPath = resolve(__dirname, "tpm-data");
```

Comment:
You might want to use [`os.tmpdir`](https://nodejs.org/api/os.html#ostmpdir) here instead.

Reply:
Good idea: [2a95570](https://github.com/mozilla/blurts-server/pull/3186/commits/2a955703d3459c925574942f708b9e853d9f4643).